Description: Cellranger count/ICGS2(AltAnalyze) analysis pipeline for running on ADT + scRNASeq data

Data type: ADTs + scRNASeq

Command: cellranger count --id=Day_21_Plasma_Cell_Precursors_Gene-Exp --transcriptome=/database/cellranger/mm10 --feature-ref=/data/salomonis2/CCHMC-Collaborations/Harinder-singh/Experiment2/Day21/feature_ref_csv_exp_2_Godhev.csv --libraries=/data/salomonis2/CCHMC-Collaborations/Harinder-singh/Experiment2/Day21/libraries_21.csv --chemistry SC3Pv3 --jobmode=lsf --maxjobs=50

Dependencies:
cellranger/5.0.1 (latest version)
bcl2fastq/2.20.0


| Parameter | Description
| ------------- | ------------- |
| id | Unique run ID used to create name of folder for results  |
| transcriptome  | path to Cell ranger comptible transcriptome reference; generated by running cellranger mkgtf on gtf of reference |
| feature-ref | path to feature reference csv file with specific columns https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/using/feature-bc-analysis#feature-ref  |
| feature-ref | path to feature reference csv file with specific columns (id,name,read,pattern,sequence,feature_type) |
| libraries | CSV file with FASTQ file paths of directories containing ADT and GEX Fastqs  |
| chemistry | Assay configuration. Should be auto-detected but sometimes need to specify manually based on cellranger output error |
| jobmode | run in single node mode or cluster mode |
| maxjobs | maximum number of job submissions sent to cluster |



Feature ref CSV format: https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/using/feature-bc-analysis#feature-ref

Creating a cellranger reference transcriptome: https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/references
cellranger mkref --genome=output_genome --fasta=input.fa --genes=input.gtf



featue-ref:
for determining the read and pattern, review https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/using/feature-bc-analysis#feature-ref and determine based on what TotalSeq assay was used for Antibody capture.
Options are:
- TotalSeq™-A
- TotalSeq™-B
- TotalSeq™-C
- CRISPR Guide Capture
*

ICGS2(AltAnalyze) pipeline:

Input: /filtered_feature_bc_matrix/matrix.mtx.gz generated by cellranger count

Dependencies:
R/3.5.0
python/2.7.5


Command:
python /data/salomonis2/software/AltAnalyze/AltAnalyze.py --runICGS yes --platform "RNASeq" --species Mm --column_method hopach --column_metric cosine --rho 0.2 --ExpressionCutoff 1 --FoldDiff 4 --SamplesDiffering 4 --restrictBy protein_coding --ChromiumSparseMatrix /data/salomonis2/CCHMC-Collaborations/Harinder-singh/Experiment2/Day21/results/Day_21_Plasma_Cell_Precursors_Gene-Exp/outs/filtered_feature_bc_matrix/matrix.mtx.gz --output /data/salomonis2/CCHMC-Collaborations/Harinder-singh/Experiment2/Day21/results/Day_21_Plasma_Cell_Precursors_Gene-Exp/outs/filtered_feature_bc_matrix --excludeCellCycle no --removeOutliers yes --expname Day_21_Plasma_Cell_Precursors_Gene-Exp



| Parameter | Description
| ------------- | ------------- |
| runICGS |   |
| platform |   |
| column_method |   |
| column_metric |   |
| rho |   |
| ExpressionCutoff |   |
| FoldDiff |   |
| SamplesDiffering |   |
| restrictBy |   |
| ChromiumSparseMatrix |   |
| output |   |
| excludeCellCycle |   |
| removeOutliers |   |
| expname |   |
